src/main/java/com/yahoo/bullet/querying/aggregations/QuantileSketchingStrategy.java

```diff
@@ -0,0 +1,85 @@
+ * This {@link Strategy} uses {@link QuantileSketch} to find distributions of a numeric field. Based on the size
+ * configured for the sketch, the normalized rank error can be determined and tightly bound.
+ */
+public class QuantileSketchingStrategy extends SketchingStrategy<QuantileSketch> {
+    private String field;
+
+    /**
+     * Constructor that requires an {@link Aggregation} and a {@link BulletConfig} configuration.
+     *
+     * @param aggregation An {@link Aggregation} with valid fields and attributes for this aggregation type.
+     * @param config The config that has relevant configs for this strategy.
+     */
+    @SuppressWarnings("unchecked")
+    public QuantileSketchingStrategy(Distribution aggregation, BulletConfig config) {
+        super(aggregation, config);
+        field = aggregation.getField();
+        sketch = getSketch(aggregation, config);
+    }
+
+    @Override
+    public void consume(BulletRecord data) {
+        Number value = extractFieldAsNumber(field, data);
+        if (value != null) {
+            sketch.update(value.doubleValue());
+        }
+    }
+
+    private static QuantileSketch getSketch(Distribution aggregation, BulletConfig config) {
+        int entries = config.getAs(BulletConfig.DISTRIBUTION_AGGREGATION_SKETCH_ENTRIES, Integer.class);
+        int rounding = config.getAs(BulletConfig.DISTRIBUTION_AGGREGATION_GENERATED_POINTS_ROUNDING, Integer.class);
+        int pointLimit = config.getAs(BulletConfig.DISTRIBUTION_AGGREGATION_MAX_POINTS, Integer.class);
+        int maxPoints = Math.min(pointLimit, aggregation.getSize());
+        BulletRecordProvider provider = config.getBulletRecordProvider();
+        if (aggregation instanceof LinearDistribution) {
+            int numberOfPoints = ((LinearDistribution) aggregation).getNumberOfPoints();
+            return new QuantileSketch(entries, rounding, aggregation.getDistributionType(), Math.min(numberOfPoints, maxPoints), provider);
+        } else if (aggregation instanceof ManualDistribution) {
+            // Limit number of points
+            List<Double> points = ((ManualDistribution) aggregation).getPoints();
+            double[] cleanedPoints = points.stream().limit(maxPoints).mapToDouble(d -> d).toArray();
+            return new QuantileSketch(entries, aggregation.getDistributionType(), cleanedPoints, provider);
+        } else if (aggregation instanceof RegionDistribution) {
+            RegionDistribution distribution = (RegionDistribution) aggregation;
+            double start = distribution.getStart();
+            double end = distribution.getEnd();
+            double increment = distribution.getIncrement();
+            return new QuantileSketch(entries, aggregation.getDistributionType(), generatePoints(start, end, increment, maxPoints, rounding), provider);
+        }
+        return null;
+    }
+
+    private static double[] generatePoints(double start, double end, double increment, int maxPoints, int rounding) {
```

Comment:
Share this with the code in QuantileSketch? Perhaps in utilities and make the signature have a function instead of the increment so we can generalize to logarithmic, exponential and whatever else point generation?